pkg/services/guardian/guardian.go

```diff
@@ -216,32 +216,36 @@ func (g *dashboardGuardianImpl) logHasPermissionResult(permission dashboards.Per
 		return hasPermission, err
 	}
 
+	var debugMessage string
 	if hasPermission {
-		g.log.Debug("User granted access to execute action", "userId", g.user.UserID, "orgId", g.orgId, "uname", g.user.Login, "dashId", g.dashId, "action", permission)
+		debugMessage = "User granted access to execute action"
 	} else {
-		g.log.Debug("User denied access to execute action", "userId", g.user.UserID, "orgId", g.orgId, "uname", g.user.Login, "dashId", g.dashId, "action", permission)
+		debugMessage = "User denied access to execute action"
 	}
 
+	namespaceID, identifier := g.user.GetNamespacedID()
+	g.log.Debug(debugMessage, "namespaceID", namespaceID, "identifier", identifier, "orgId", g.orgId, "uname", g.user.GetLogin(), "dashId", g.dashId, "action", permission)
+
 	return hasPermission, err
 }
 
 func (g *dashboardGuardianImpl) checkACL(permission dashboards.PermissionType, acl []*dashboards.DashboardACLInfoDTO) (bool, error) {
-	orgRole := g.user.OrgRole
 	teamACLItems := []*dashboards.DashboardACLInfoDTO{}
 
 	for _, p := range acl {
+		userID, err := identity.IntIdentifier(g.user.GetNamespacedID())
```

Comment:
not guaranteed to be a user ID

Reply:
check namespace, if API key set user ID to 0 if you want to keep legacy support for API keys.

Reply:
Also allow render service with user ID 0 otherwise dashboards won't be rendered to PDF

Reply:
This function has been removed after merge.